nemo_skills/evaluation/metrics/base.py

```diff
@@ -29,6 +30,92 @@ def get_metrics(self):
     def reset(self):
         pass
 
+    @abc.abstractmethod
+    def get_prediction_results(self, prediction):
+        pass
+
+    def get_majority_at_k(
+        self, agg_mode_dict, predicted_answers, pred_keys=None, predictions=None, prediction_results=None
```

Comment:
I would probably also remove pred_keys and always assume that whatever is returned in get_prediction_results is what we should use. I think you only use pred_keys in math metrics and if metrics. In math metrics, from what I can see it might be possible to just completely remove it? Maybe I'm missing something, but seems that it will already contain exactly the right keys based on what's available for judge/sympy metrics and then we can just aggregate over all of them. And for the if metrics, to support that use-case, I'd instead change the self._get_best_prediction signature to also accept full predictions list and then it will have access to any parameters it needs. Let me know if I miss something here

Reply:
I thought this metric might be used by other benchmarks as well so I kept it in the main file. If not, I'll just move it to the math metric file.

Reply:
Actually there is another one where it makes sense, the https://github.com/NVIDIA/NeMo-Skills/blob/main/nemo_skills/evaluation/metrics/answer_judgement_metrics.py. I think you missed it in this PR, can you please change that class accordingly? And in that case we can keep it in the base metrics.

There is also arena_metrics and mtbench_metrics that would need changes I guess?